topics/basics/testing_plugins/integration_tests/integration_tests_intro.md

```diff
@@ -0,0 +1,262 @@
+
+Add the following configuration to your `build.gradle.kts`:
+```
+tasks.test {
+    dependsOn("buildPlugin")
+    systemProperty("path.to.build.plugin", tasks.buildPlugin.get().archiveFile.get().asFile.absolutePath)
+    useJUnitPlatform()
+}
+```
+
+This configuration does the following:
+* Makes the test task depend on `buildPlugin`, ensuring your plugin is built before tests run.
+* Sets the `path.to.build.plugin` system property to point to your plugin distribution file.
+* Enables JUnit Platform for test execution.
+
+## Creating the first test
+
+Now that we've completed the configuration, let's write our first integration test, which will:
+* Start the IDE with our plugin installed.
+* Wait for all background processes to complete.
+* Perform a shutdown.
+
+Create a new Kotlin file in `src/test/kotlin` with the following code:
+
+```kotlin
+class PluginTest {
+   @Test
+   fun simpleTestWithoutProject() {
+      Starter.newContext(testName = "testExample", TestCase(IdeProductProvider.IC, projectInfo = NoProject).withVersion("2024.3")).apply {
+         val pathToPlugin = System.getProperty("path.to.build.plugin")
+         PluginConfigurator(this).installPluginFromPath(Path(pathToPlugin))
+      }.runIdeWithDriver().useDriverAndCloseIde {
+      }
+   }
+}
+```
+
+Let's break down each part of the test:
+### 1. Context creation
+`Starter.newContext(testName = "testExample", TestCase(IdeProductProvider.IC, projectInfo = NoProject).withVersion("2024.3"))`
+
+The Context object stores IDE runtime configuration:
+* IDE type (e.g., IntelliJ Community, PhpStorm, GoLand).
+* IDE version (2024.3 in this example).
+* Project configuration (using NoProject for this example).
+* Custom VM options, paths, and SDK settings.
+
+The `testName` parameter defines the folder name for test artifacts, which is useful when running multiple IDE instances in a single test. We're using IntelliJ IDEA Community Edition version 2024.3, and we're starting the IDE without any project, so the welcome screen will be shown.
+### 2. Plugin installation
+```kotlin
+.apply {
+   val pathToPlugin = System.getProperty("path.to.build.plugin")
+   PluginConfigurator(this).installPluginFromPath(Path(pathToPlugin))
+}
+```
+
+This step configures plugin installation using the plugin path we defined in our Gradle configuration.
+
+Note: `PluginConfigurator` can install plugins from local paths or Marketplace.
+
+### 3. IDE life cycle management
+```kotlin
+.runIdeWithDriver().useDriverAndCloseIde {
+}
+```
+
+These two methods:
+* Start the IDE instance (`runIdeWithDriver`).
+* Shut down the IDE (`useDriverAndCloseIde`).
+
+The empty lambda is used for IDE interactions (`useDriverAndCloseIde`).
+
+Note: When you run the test for the first time, it may take longer than expected since it needs to download the IDE. Subsequent runs will be faster, using the cached IDE version.
```

Comment:
use real callouts? https://plugins.jetbrains.com/docs/intellij/sdk-style.html#notes-and-callouts